packages/hardhat-core/src/internal/hardhat-network/provider/output.ts

```diff
@@ -178,8 +178,13 @@ export function getRpcBlock(
     timestamp: numberToRpcQuantity(new BN(block.header.timestamp)),
     transactions,
     uncles: block.uncleHeaders.map((uh: any) => bufferToRpcData(uh.hash())),
-    baseFeePerGas: numberToRpcQuantity(block.header.baseFeePerGas ?? 0),
   };
+
+  if (block.header.baseFeePerGas) {
```

Comment:
I think we always need to return it if london is active. We should check what geth does.

Reply:
i.e. what should the `else` be?

Reply:
My bad, just checked with Geth and it doesn't show this field for non-eip-1559 blocks.